scripts/imgtool/keys/ecdsa.py

```diff
@@ -0,0 +1,102 @@
+"""
+ECDSA key management
+"""
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.hashes import SHA256
+
+from .general import KeyClass
+
+class ECDSAUsageError(Exception):
+    pass
+
+class ECDSA256P1Public(KeyClass):
+    def __init__(self, key):
+        self.key = key
+
+    def shortname(self):
+        return "ecdsa"
+
+    def _unsupported(self, name):
+        raise ECDSAUsageError("Operation {} requires private key".format(name))
+
+    def _get_public(self):
+        return self.key
```

Comment:
I would have gone for using `self.key` directly below.

Reply:
The method is overridden below, and returns something different. Using self.key wouldn't work there.